format/spec.md

```diff
@@ -241,7 +241,13 @@ Struct evolution requires the following rules for default values:
 
 #### Column Projection
 
-Columns in Iceberg data files are selected by field id. The table schema's column names and order may change after a data file is written, and projection must be done using field ids. If a field id is missing from a data file, its value for each row should be `null`.
+Columns in Iceberg data files are selected by field id. The table schema's column names and order may change after a data file is written, and projection must be done using field ids.
+
+Values for Field ids which are not present in a data file must be resolved according the following rules:
+
+* Return the value from partition metadata if an [Identity Transform](#partition-transforms) exists for the field. 
+* Return the default value as defined in [Default values](#default-values) if it exists.
```

Comment:
I think we can be clear here that in cases where the column is not present in a data file (after applying name mappings) the `initial-default` must be used.

Reply:
I split the name mapping into a different step.